### ISSUE TITLE
monthly contributions contributing again after browser restart when AC is disabled

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#10006]
## Description 

If you have auto contribute disabled but you have recurring monthly tips set up, your monthly tips contribute as expected at the designated date/time. However, if you restart your browser after they've contributed, on restart they will contribute again.

## Steps to Reproduce

   1. Clean profile w/ 1.10.x, launch with something like `/Applications/Brave\ Browser\ Dev.app/Contents/MacOS/Brave\ Browser\ Dev --enable-logging=stderr --vmodule=*rewards*=6 --log-level=2 --rewards=staging=true,reconcile-interval=5 --brave-ads-staging` (can use 10 instead of 5 if you want additional time)
   2. Enable Rewards.
   3. Claim UGP.
   4. Disable auto contribute.
   5. Set up recurring tips (I did 3 - 1 BAT to 3zsistemi.si, 5 BAT to github.com/laurenwags, and 1 BAT to yachtcaptain23.github.io)
   6. Wait for tips to contribute.
   7. Once they're done, restart the browser (be sure you have enough time before your next "scheduled" contribution time).

When retesting this, QA should also check with AC on + recurring tips as well as just AC and no recurring tips.

## Actual result:

Monthly tips contribute again after browser restart (pretty much immediately)

## Expected result:
Monthly tips should not contribute again until scheduled date/time

## Reproduces how often: 

easily

## Brave version (brave://version info)

Brave | 1.10.76 Chromium: 81.0.4044.138 (Official Build) beta (64-bit)
-- | --
Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS | macOS Version 10.14.6 (Build 18G3020)

## Version/Channel Information:

- Can you reproduce this issue with the current release? no, does not occur on 1.9.x
- Can you reproduce this issue with the beta channel? yes
- Can you reproduce this issue with the dev channel? yes
- Can you reproduce this issue with the nightly channel? unsure, probably

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @NejcZdovc @brave/legacy_qa @rebron

**Comment by GeetaSarvadnya** [comment]
Reproduced in 
```

Brave | 1.10.77 Chromium: 81.0.4044.138 (Official Build) dev (64-bit)
-- | --
Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS | Windows 10 OS Version 1803 (Build 17134.1006)

```

**Comment by NejcZdovc** [comment]
@GeetaSarvadnya it was not merged in 1.10 yet

**Comment by LaurenWags** [comment]
Verified passed with


Brave | 1.10.82 Chromium: 81.0.4044.138 (Official Build) dev (64-bit)
-- | --
Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS | macOS Version 10.14.6 (Build 18G3020)

* Verified test plan as written from description. Confirmed after browser restart my recurring tips were not contributed again. 

Also verified the below scenarios:

Scenario 1: AC and recurring tips set up
1. Clean profile, launch with something like /`Applications/Brave\ Browser\ Dev.app/Contents/MacOS/Brave\ Browser\ Dev --enable-logging=stderr --vmodule=*rewards*=6 --log-level=2 --rewards=staging=true,reconcile-interval=7 --brave-ads-staging`
2. Enable Rewards, claim UGP.
3. Add 3 recurring tips.
4. Add 1 site to AC table (due to #10058)
5. Wait for AC and recurring tips to be contributed.
6. Add another site (just one due to above issue) to AC table.
6. Restart the browser (be sure you have enough time before your next "scheduled" contribution time).
--> Confirmed AC and recurring tips did not contribute a second time after browser restart.

Scenario 2: only AC set up, no recurring tips
1. Clean profile, launch with something like /`Applications/Brave\ Browser\ Dev.app/Contents/MacOS/Brave\ Browser\ Dev --enable-logging=stderr --vmodule=*rewards*=6 --log-level=2 --rewards=staging=true,reconcile-interval=5 --brave-ads-staging`
2. Enable Rewards, claim UGP.
3. Add 1 site to AC table (due to #10058)
4. Wait for AC to be contributed.
5. Add another site to AC list (just one due to above issue).
6. Restart the browser (be sure you have enough time before your next "scheduled" contribution time).
--> Confirmed AC did not contribute a second time after browser restart.
